src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspace.cs

```diff
@@ -89,27 +103,33 @@ internal override ValueTask TryOnDocumentClosedAsync(DocumentId documentId, Canc
         return this.ProjectSystemProjectFactory.ApplyChangeToWorkspaceAsync(
             async w =>
             {
-                // TODO(cyrusn): This only works for normal documents currently.  We'll have to rethink how things work
-                // in the world if we ever support additionalfiles/editorconfig in our language server.
-                var document = w.CurrentSolution.GetDocument(documentId);
+                var textDocument = w.CurrentSolution.GetDocument(documentId) ?? w.CurrentSolution.GetAdditionalDocument(documentId);
```

Comment:
@jasonmalinowski  just like DocIds state if they are SG docs or not, i think docIds should state if they are additional documents or not.  I *really* hate this sort of appraoch where we just get a doc id in and "guess" what it is.  Everything from end-to-end should be strongly knowledgeable here.  So, for example, the caller should know if its talking about normal docs or additional docs, and it should call a dedicated TryOnDocumentClosedAsync and TryOnAdditionalDocuemntClosedAsync, versus going through one path which just guesses as to what's going on.